Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py

```diff
@@ -358,6 +361,26 @@ def set(self, issue, assignee=None, opened=None, why=None, project=None, compone
             issue._component = component
             issue._version = version
 
+        if keywords is not None:
+            for keyword in keywords + issue.keywords:
+                if keyword not in self._invalid_keywords and keyword not in self._keywords:
+                    candidates = self.client.keywords_for_name(keyword)
+                    for candidate in candidates:
+                        self._keywords[candidate.name] = candidate
+                if keyword in self._keywords:
+                    continue
+                self._invalid_keywords.add(keyword)
+                raise ValueError("'{}' is not a valid keyword".format(keyword))
```

Comment:
It looks like we raise the exception the first time we add keyword to `_invalid_keywords`, that means we are always checking against empty `_invalid_keywords` on line 366. If that's the case, do we still need this variable?

Reply:
Checking for keywords is actually a very expensive network call. Presumably, our caller will be handling this exception, and in that case, we can save an expensive network call if our caller re-attempts the same invalid keyword in a single program. That's actually a relatively common event in practice, where a program is likely to have keywords constructed from variables and then re-used on multiple issues.